lib/module.js

```diff
@@ -95,38 +96,83 @@ export default function SentryModule (moduleOptions) {
     }
   }
 
+  if (options.lazy) {
+    const defaultLazyOptions = {
+      injectMock: true,
+      injectLoadHook: false,
+      mockApiMethods: true,
+      chunkName: 'sentry',
+      webpackPrefetch: false,
+      webpackPreload: false
+    }
+
+    options.lazy = Object.assign({}, defaultLazyOptions, options.lazy)
+  }
+
+  const apiMethods = await getBrowserApiMethods()
```

Comment:
Not feeling great about this one since `mockApiMethods` is part of the lazy options but `apiMethods` is also used when not using `lazy`. So you can only customize what is mocked when also using `lazy`.

Reply:
:+1:

Reply:
(I have not addressed this one, even though Github thinks I did)

What do you think about removing the `mockApiMethods` option and always mocking all methods if `injectMock` is enabled? It feels to me too eager to add `mockApiMethods` but maybe that's because I don't see the use case for only enabled mocking of some methods.

Reply:
Its a (micro-)optimization for your bundle size, ie why inject those method names into your bundle when you know you wont ever use them.

Reply:
It's just a dozen of function names so I don't feel like it's worth adding an option for that.

And if kept, it would have to be refactored to be a global option and apply also to non-lazy mock.

Reply:
I implemented this for lazy only because only lazy really needs the mock/proxy methods. Also, Im a strong believer that any module should provide full configurability as to which options to include that affect bundle size. This single change might seem small, but its already 300bytes uncompressed that needs to be unzipped + parsed. If Im using 10 modules which would be this easy-going, thats 3KB. All in all its not about this single optimization but about all of these small optimizations combined that will provide a benefit.

In all other cases I actually would prefer to just not provide a mock. Eg if a user decides to not include sentry on the server then he is responsible himself to just not include it. Adding a mock for that is just too much user friendliness that only benefits bad practices. Or I might be misinterpreting why the mock existed originally.

Reply:
I think the mock was more for development vs. production cases. You would have mocked instance during development so that no issue would be reported.

But I can't recall what was the issue with setting `initialize: false` instead which should be able to satisfy that required too. Maybe @simllll can remind us his perspective on that.